cc/README.md

```diff
@@ -9,115 +10,189 @@ build system. We have experienced build issues with the latest version of Bazel,
-bazel-bin/cc/main --model=$MODEL_PATH --mode=selfplay
+bazel build -c opt cc:selfplay
+bazel-bin/cc/selfplay \
+  --model=saved_models/000990-cormorant.pb \
+  --num_readouts=160 \
+  --parallel_games=1 \
+  --output_dir=data/selfplay \
+  --holdout_dir=data/holdout \
+  --sgf_dir=sgf
 ```
 
-To run Minigo with a 19x19 model:
+#### cc:eval
+
+Evaluates the performance of two models by playing them against each other over
+multiple games in parallel.
 
 ```shell
-bazel build -c opt cc:main
-bazel-bin/cc/main --model=$MODEL_PATH --mode=selfplay
+bazel build -c opt cc:eval
+bazel-bin/cc/eval \
+  --model=saved_models/000990-cormorant.pb \
+  --model_two=saved_models/000990-cormorant.pb \
+  --num_readouts=160 \
+  --parallel_games=32 \
+  --sgf_dir=sgf
 ```
 
-The Minigo binary has a lot of command line arguments that configure its
-behavior, run it with --helpshort to see the full list.
+#### cc:gtp
+
+Play using the GTP protocol. This is also the binary we recommend using as a
+backend for Minigui (see `minigui/README.md`).
 
 ```shell
-bazel-bin/cc/main --helpshort
+bazel build -c opt cc:gtp
+bazel-bin/cc/gtp \
+  --model=saved_models/000990-cormorant.pb \
+  --num_readouts=160
 ```
 
-## Design
+#### cc:puzzle
 
-The general structure of the C++ code tries to follow the Python code where
-appropriate, however a lot of the implementation details are different for
-performance reasons. In particular, the logic that handles the board state
-strives for a small memory footprint and eschews data structures more
-sophisticated than a simple array in an effort to minimize memory allocations
-and maximize cache locality of data. In doing so, the implementation is kept
-fairly simple (there is no need for a LibertyTracker) and at the time of writing
-performance of the Position code was more than 450x that of its Python
-counterpart.
+Loads all SGF files found in the given `sgf_dir` and tries to predict the move
+made at each position in each game. After all games are processed, prints
+summary stats to about how many moves were correctly predicted.
 
-## Inference engines
+```shell
+bazel build -c opt cc:puzzle
+bazel-bin/cc/puzzle \
+  --model=saved_models/000990-cormorant.pb \
+  --num_readouts=160 \
+  --sgf_dir=puzzle_sgf
+```
 
-C++ Minigo currently supports three separate engines for performing inference:
+## Running the unit tests
 
- - tf: peforms inference using the TensorFlow libraries built by
-   `cc/configure_tensorflow.sh`.
- - lite: performs inference using TensorFlow Lite, which runs in software on
-   the CPU.
+Minigo's C++ unit tests operate on both 9x9 and 19x19, and some tests are only
+enabled for a particular board size. Consequently, you must run the tests
+multiple times:
 
-The Compilation and linking of these engines into the `//cc:main` binary is
-controlled by the Bazel defines `--define=tf=<0,1>` and `--define=lite=<0,1>`.
+```shell
+bazel test --define=board_size=9 cc/...  &&  bazel test cc/...
```

Comment:
maybe either 
1. call out that --define=board_size=19 is the default in the text above
2. explicitly write it here.

Reply:
1 it is.

The defines don't work how a sane person would expect, so --define=board_size=19 would actually just be completely ignored :P

Reply:
.... that's amazing.